examples/federated-learning/run.py

```diff
@@ -131,74 +80,124 @@ class DataOwner:
     build_update_step: `Callable`, the function used to construct
                        a local federated learning update.
   """
-
-  BATCH_SIZE = 30
-
-  def __init__(self, player_name, local_data_file, build_update_step):
+  def __init__(self, player_name, local_data_file, loss):
     self.player_name = player_name
     self.local_data_file = local_data_file
-    self._build_update_step = build_update_step
+    self.loss = loss
+
+    self.dataset = iter(build_data_pipeline(player_name=self.player_name))
 
-  def _build_data_pipeline(self):
-    """Build local data pipeline for federated DataOwners."""
-    def normalize(image, label):
-      image = tf.cast(image, tf.float32) / 255.0
-      return image, label
+@tfe.local_computation
+def update_model(model_owner, *grads):
```

Comment:
@jvmancuso @yanndupis suggested that these methods should not live on the classes because we were trying to make it more natural how a data scientist would actually program a training workflow. Personally, I'm not sure it makes a difference and putting the calls on the classes makes the calls a little less verbose by removing `player_name=` part of it.

Reply:
maybe it's worth a shot. there's less point in having the `DataOwner` and `ModelOwner` classes in this case though. for the model owner the `player_name=` could be removed by passing it directly to `local_computation`.

i'm wondering if `local_computation` is the right construct to use actually. one issue is that it's protocol dependent, which is why you had to modify Pond, although we should only care about Pond for the secure aggregation part. maybe there's a way to unity these uses of `local_computation`, but also wondering if it wouldn't be just as good simply to use eg `with tf.device(model_owner.device_name):` here?

Reply:
You need pond for securing the grads when they come from the data owner too don't you? And the updating the grads on the model owner's model?

Reply:
> You need pond for securing the grads when they come from the data owner too don't you?

Absolutely! Sharing should happen with Pond.

>  And updating the grads on the model owner's model needs to be secured too?

The aggregated gradients should be revealed to the model owner, but the actual updating could/should happen independently of Pond.